### ISSUE TITLE
Compatible NNLib version

### ISSUE THREAD

**Issue by mschauer** [vincentmolin/WGANGP.jl#5]
Hi Vincent, what was the compatible NNLib version? Does master from here https://github.com/vincentmolin/NNlib.jl work?

**Comment by mschauer** [comment]
Could it be that this fixed itself somehow, can you try if everything runs with NNlib v0.8.9 (the latest version)? Then we can register the package right away @vincentmolin

**Comment by vincentmolin** [comment]
Test fails with Flux 0.13.4 and NNlib 0.8.9, same error as before. I think the error was due to a change on Flux master between 0.12.x and 0.13.0, the working branch is from 0.13.0-DEV. I'll look through the convo from earlier this year

**Comment by mschauer** [comment]
For me it works with Google Colab and 
```
Status `~/.julia/environments/v1.7/Manifest.toml`
  [621f4979] AbstractFFTs v1.2.1
  [79e6a3ab] Adapt v3.4.0
  [ec485272] ArnoldiMethod v0.2.0
  [4fba245c] ArrayInterface v6.0.22
  [30b0a656] ArrayInterfaceCore v0.1.17
  [13072b0f] AxisAlgorithms v1.0.1
  [39de3d68] AxisArrays v0.4.6
  [ab4f0b2a] BFloat16s v0.2.0
  [fbb218c0] BSON v0.3.5
  [6e4b80f9] BenchmarkTools v1.3.1
  [9e28174c] BinDeps v1.0.2
  [b99e7846] BinaryProvider v0.5.10
  [e1450e63] BufferedStreams v1.1.0
  [fa961155] CEnum v0.4.2
  [052768ef] CUDA v3.12.0
  [49dc2e85] Calculus v0.5.1
  [aafaddc9] CatIndices v0.2.2
  [8e462317] CausalInference v0.8.0
  [082447d4] ChainRules v1.44.1
  [d360d2e6] ChainRulesCore v1.15.3
  [9e997f8a] ChangesOfVariables v0.1.4
  [aaaa29a8] Clustering v0.14.2
  [944b1d66] CodecZlib v0.7.0
  [35d6a980] ColorSchemes v3.19.0
  [3da002f7] ColorTypes v0.11.4
  [c3611d14] ColorVectorSpace v0.9.9
  [5ae59095] Colors v0.12.8
  [861a8166] Combinatorics v1.0.2
  [bbf7d656] CommonSubexpressions v0.3.0
  [34da2185] Compat v4.1.0
  [ed09eef8] ComputationalResources v0.3.2
  [8f4d0f93] Conda v1.7.0
  [d38c429a] Contour v0.6.2
  [150eb455] CoordinateTransformations v0.6.2
  [dc8bdbbb] CustomUnitRanges v1.0.2
  [9a962f9c] DataAPI v1.10.0
  [124859b0] DataDeps v0.7.9
  [864edb3b] DataStructures v0.18.13
  [e2d170a0] DataValueInterfaces v1.0.0
  [b429d917] DensityInterface v0.4.0
  [163ba53b] DiffResults v1.0.3
  [b552c78f] DiffRules v1.11.0
  [b4f34e82] Distances v0.10.7
  [31c24e10] Distributions v0.25.66
  [ffbed154] DocStringExtensions v0.9.1
  [fa6b7ba4] DualNumbers v0.6.8
  [e2ba6199] ExprTools v0.1.8
  [411431e0] Extents v0.1.1
  [c87230d0] FFMPEG v0.4.1
  [4f61f5a4] FFTViews v0.3.2
  [7a1cc6ca] FFTW v1.5.0
  [5789e2e9] FileIO v1.15.0
  [1a297f60] FillArrays v0.13.2
  [53c48c17] FixedPointNumbers v0.8.4
  [587475ba] Flux v0.13.5
  [59287772] Formatting v0.4.2
  [f6369f11] ForwardDiff v0.10.32
  [d9f16b24] Functors v0.3.0
  [0c68f7d7] GPUArrays v8.4.2
  [46192b85] GPUArraysCore v0.1.1
  [61eb1bfa] GPUCompiler v0.16.3
  [28b8d3ca] GR v0.66.0
  [92fee26a] GZip v0.5.1
  [cf35fbd7] GeoInterface v1.0.1
  [5c1252a2] GeometryBasics v0.4.3
  [a2bd30eb] Graphics v1.1.2
  [86223c79] Graphs v1.7.1
  [42e2da0e] Grisu v1.0.2
  [f67ccb44] HDF5 v0.16.10
  [cd3eb016] HTTP v1.2.1
  [34004b35] HypergeometricFunctions v0.3.11
  [7073ff75] IJulia v1.23.3
  [7869d1d1] IRTools v0.4.6
  [615f187c] IfElse v0.1.1
  [2803e5a7] ImageAxes v0.6.10
  [c817782e] ImageBase v0.1.5
  [f332f351] ImageContrastAdjustment v0.3.10
  [a09fc81d] ImageCore v0.9.4
  [51556ac3] ImageDistances v0.2.16
  [6a3955dd] ImageFiltering v0.7.1
  [82e4d734] ImageIO v0.6.6
  [6218d12a] ImageMagick v1.2.2
  [bc367c6b] ImageMetadata v0.9.8
  [787d08f9] ImageMorphology v0.3.2
  [2996bd0c] ImageQualityIndexes v0.3.3
  [80713f31] ImageSegmentation v1.7.0
  [4e3cecfd] ImageShow v0.3.6
  [02fcd773] ImageTransformations v0.9.5
  [916415d5] Images v0.25.2
  [9b13fd28] IndirectArrays v1.0.0
  [d25df0c9] Inflate v0.1.2
  [83e8ac13] IniFile v0.5.1
  [1d092043] IntegralArrays v0.1.5
  [7d512f48] InternedStrings v0.7.0
  [a98d9a8b] Interpolations v0.14.4
  [8197267c] IntervalSets v0.7.1
  [3587e190] InverseFunctions v0.1.7
  [92d709cd] IrrationalConstants v0.1.1
  [c8e1da08] IterTools v1.4.0
  [82899510] IteratorInterfaceExtensions v1.0.0
  [033835bb] JLD2 v0.4.22
  [692b3bcd] JLLWrappers v1.4.1
  [682c06a0] JSON v0.21.3
  [0f8b85d8] JSON3 v1.9.5
  [b835a17e] JpegTurbo v0.1.1
  [929cbde3] LLVM v4.14.0
  [b964fa9f] LaTeXStrings v1.3.0
  [23fbe1c1] Latexify v0.15.16
  [8cdb02fc] LazyModules v0.3.1
  [2ab3a3ac] LogExpFunctions v0.3.17
  [e6f89c97] LoggingExtras v0.4.9
  [23992714] MAT v0.10.3
  [eb30cadb] MLDatasets v0.5.15
  [f1d291b0] MLUtils v0.2.1
  [1914dd2f] MacroTools v0.5.9
  [dbb5928d] MappedArrays v0.4.1
  [739be429] MbedTLS v1.1.3
  [442fdcdd] Measures v0.3.1
  [626554b9] MetaGraphs v0.7.1
  [e1d29d7a] Missings v1.0.2
  [e94cdb99] MosaicViews v0.3.3
  [872c559c] NNlib v0.8.9
  [a00861dc] NNlibCUDA v0.2.4
  [77ba4419] NaNMath v1.0.1
  [b8a86587] NearestNeighbors v0.4.11
  [f09324ee] Netpbm v1.0.2
  [6fe1bfb0] OffsetArrays v1.12.7
  [52e1d378] OpenEXR v0.3.2
  [3bd65402] Optimisers v0.2.9
  [bac558e1] OrderedCollections v1.4.1
  [90014a1f] PDMats v0.11.16
  [f57f5aa1] PNGFiles v0.3.16
  [5432bcbf] PaddedViews v0.5.11
  [d96e819e] Parameters v0.12.3
  [69de0a69] Parsers v2.3.2
  [fbb45041] Pickle v0.3.2
  [eebad327] PkgVersion v0.1.1
  [ccf2f8ad] PlotThemes v3.0.0
  [995b91a9] PlotUtils v1.3.0
  [91a5bcdd] Plots v1.31.7
  [21216c6a] Preferences v1.3.0
  [33c8b6b6] ProgressLogging v0.1.4
  [92933f4c] ProgressMeter v1.7.2
  [3349acd9] ProtoBuf v0.11.5
  [4b34888f] QOI v1.0.0
  [1fd47b50] QuadGK v2.4.2
  [94ee1d12] Quaternions v0.5.6
  [74087812] Random123 v1.6.0
  [e6cf234a] RandomNumbers v1.5.3
  [b3c3ace0] RangeArrays v0.3.2
  [c84ed2f1] Ratios v0.4.3
  [c1ae055f] RealDot v0.1.0
  [3cdcf5f2] RecipesBase v1.2.1
  [01d81517] RecipesPipeline v0.6.3
  [189a3867] Reexport v1.2.2
  [dee08c22] RegionTrees v0.3.2
  [05181044] RelocatableFolders v0.3.0
  [ae029012] Requires v1.3.0
  [79098fc4] Rmath v0.7.0
  [6038ab10] Rotations v1.3.1
  [6c6a2e73] Scratch v1.1.1
  [605ecd9f] ShowCases v0.1.0
  [992d4aef] Showoff v1.0.3
  [777ac1f9] SimpleBufferStream v1.1.0
  [699a6c99] SimpleTraits v0.9.4
  [47aef6b3] SimpleWeightedGraphs v1.2.1
  [45858cf5] Sixel v0.1.2
  [b85f4697] SoftGlobalScope v1.1.0
  [a2af1166] SortingAlgorithms v1.0.1
  [276daf66] SpecialFunctions v2.1.7
  [cae243ae] StackViews v0.1.1
  [aedffcd0] Static v0.7.6
  [90137ffa] StaticArrays v1.5.3
  [1e83bf80] StaticArraysCore v1.1.0
  [82ae8749] StatsAPI v1.5.0
  [2913bbd2] StatsBase v0.33.21
  [4c63d2b9] StatsFuns v1.0.1
  [5e0ebb24] Strided v1.2.3
  [69024149] StringEncodings v0.3.5
  [09ab397b] StructArrays v0.6.11
  [856f2bd8] StructTypes v1.9.0
  [3783bdb8] TableTraits v1.0.1
  [bd369af6] Tables v1.7.0
  [9ac5f52a] Tectonic v0.8.0
  [899adc3e] TensorBoardLogger v0.1.19
  [62fd8b95] TensorCore v0.1.1
  [731e570b] TiffImages v0.6.0
  [b4f28e30] TikzGraphs v1.4.0
  [37f6aa50] TikzPictures v3.4.2
  [06e1c1a7] TiledIteration v0.3.1
  [a759f4b9] TimerOutputs v0.5.20
  [3bb67fe8] TranscodingStreams v0.9.6
  [9d95972d] TupleTools v1.3.0
  [30578b45] URIParser v0.4.1
  [5c2747f8] URIs v1.4.0
  [3a884ed6] UnPack v1.0.2
  [1cfade01] UnicodeFun v0.4.1
  [41fe7b60] Unzip v0.1.2
  [81def892] VersionParsing v1.3.0
  [1511ac5a] WGANGP v0.1.0 `[https://github.com/vincentmolin/WGANGP.jl.git#main`](https://github.com/vincentmolin/WGANGP.jl.git#main%60)
  [efce3f68] WoodburyMatrices v0.5.5
  [c2297ded] ZMQ v1.2.1
  [a5390f91] ZipFile v0.10.0
  [e88e6eb3] Zygote v0.6.43
  [700de1a5] ZygoteRules v0.2.2
```

**Comment by mschauer** [comment]
Solved, #f6a261fc5d043053203bdfabbff0aad67f099cab
